indico/modules/events/registration/controllers/display.py

```diff
@@ -447,6 +450,18 @@ def get_file_context(self):
         return 'event', self.event.id, 'regform', self.regform.id, 'registration'
 
 
+class RHUploadRegistrationPicture(RHUploadRegistrationFile):
+    """Upload a picture from a registration form."""
+
+    def _save_file(self, file, stream):
+        from indico.modules.files.schemas import FileSchema
+        context = self.get_file_context()
+        content_type = mimetypes.guess_type(file.filename)[0] or file.mimetype or 'application/octet-stream'
+        resized_image_bytes = resize_uploaded_registration_picture(stream, 1000)
+        f = File.create_from_stream(resized_image_bytes, file.filename, content_type, context)
+        return FileSchema().jsonify(f), 201
```

Comment:
Can't you just resize the image and then call the original method from the superclass with the new stream? That way you won't have to duplicate all that code here.

Reply:
Ah yes I can do that!